src/vs/workbench/services/workspaces/browser/abstractWorkspaceEditingService.ts

```diff
@@ -204,6 +206,9 @@ export abstract class AbstractWorkspaceEditingService implements IWorkspaceEditi
 		if (path) {
 			try {
 				await this.saveWorkspaceAs(untitledWorkspace, path);
+
+				// Set trust for the workspace file
+				this.trustWorkspaceConfiguration(path);
```

Comment:
I think if you move this code into the `saveWorkspaceAs` method, a lot of the changes become much simpler?

Reply:
That was my original plan but I ran into some issues. Will give it another try.

Reply:
Moved the call into `this.saveWorkspaceAs`.